packages/transaction-controller/src/utils/utils.ts

```diff
@@ -31,7 +31,7 @@ const NORMALIZERS: { [param in keyof TransactionParams]: any } = {
     addHexPrefix(maxPriorityFeePerGas),
   estimatedBaseFee: (maxPriorityFeePerGas: string) =>
     addHexPrefix(maxPriorityFeePerGas),
-  type: (type: string) => (type === '0x0' ? '0x0' : undefined),
+  type: (type: string) => addHexPrefix(type),
```

Comment:
In a future PR, we could go one step further and always derive a `type` even if one wasn't provided by looking at whether there is `gasPrice` or `maxFeePerGas` etc.

That's what we already do on submission using `isEIP1559Transaction`.

Perhaps worth confirming by testing an older commit with the extension controller, but my understanding is if a dapp transaction doesn't specify type, it won't be in the parameters either.

Reply:
I think this would be a good addition yeah! 👍